File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/RecordAttValue.java
package ru.citeck.ecos.records3.record.op.atts.service.value.impl;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.records3.record.op.atts.dto.RecordAtts;
import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue;

public class RecordAttValue implements AttValue {

    private final RecordAtts atts;

    public RecordAttValue(RecordAtts atts) {
        this.atts = atts;
    }

    @Override
    public String getId() {
        return atts.getId().toString();
    }

    @Override
    public String asText() {
        return atts.getId().toString();
    }

    @Override
    public Object getAtt(@NotNull String name) {
        return atts.get(name);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/RecordAttValue.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.impl

import ru.citeck.ecos.records3.record.op.atts.dto.RecordAtts
import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

class RecordAttValue(val atts: RecordAtts) : AttValue {

    override fun getId(): String? {
        return atts.getId().toString()
    }

    override fun asText(): String? {
        return atts.getId().toString()
    }

    override fun getAtt(name: String): Any? {
        return atts.getAtt(name)
    }
}
